drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -200,139 +187,86 @@ static int get_temperature(s16_t *tvp)
 static void measure_temperature(struct k_work *work)
 {
 	s16_t temperature = 0;
-	s16_t diff;
+	s16_t diff = 0;
 	bool started = false;
-	int key;
 	int rc;
 
 	rc = get_temperature(&temperature);
 
-	key = irq_lock();
-
 	if (rc != 0) {
 		/* Temperature read failed: retry later */
```

Comment:
Relates to the other comment: Do you expect the temperature read to ever fail?  How often?

As written, if the read always fails, the clock will never be calibrated.  Perhaps a better workaround would be to set `calib_skip_cnt = 0` and keep going.

Reply:
done